server/graphql/cache.js

```diff
@@ -1,176 +1,27 @@
-import { difference, isEqual } from 'lodash';
-
-import { TransactionKind } from '../constants/transaction-kind';
 import { md5 } from '../lib/utils';
 
-export const purgeCacheForCollectiveOperationNames = [
-  'CollectivePage',
-  'ContributePage',
-  'BudgetSection',
-  'BudgetSectionWithHost',
-  'UpdatesSection',
-  'TransactionsSection',
-  'TransactionsPage',
-  'ExpensesPage',
-  'CollectiveBannerIframe',
-  'CollectiveCover',
-  'RecurringContributions',
-  'ContributionsSectionStatic',
-  'Members_Users',
-  'Members_Organizations',
-];
-
-export function checkSupportedVariables(req, variableNames) {
-  if (!req.body.variables) {
-    return false;
-  }
-  // Check missing variables
-  if (difference(variableNames, Object.keys(req.body.variables)).length > 0) {
-    return false;
-  }
-  // Check extra variables
-  if (difference(Object.keys(req.body.variables), variableNames).length > 0) {
-    return false;
-  }
-
-  return true;
-}
-
-function getCacheKeyForBudgetOrTransactionsSections(req, queryHash) {
-  if (
-    req.body.variables.kind &&
-    !isEqual(req.body.variables.kind.sort(), [
-      TransactionKind.ADDED_FUNDS,
-      TransactionKind.BALANCE_TRANSFER,
-      TransactionKind.CONTRIBUTION,
-      TransactionKind.EXPENSE,
-      TransactionKind.PLATFORM_TIP,
-    ])
-  ) {
+export function getGraphqlCacheProperties(req) {
+  if (req.remoteUser) {
     return;
   }
-  return `${req.body.operationName}_${queryHash}_${req.body.variables.slug}`;
-}
 
-export function getGraphqlCacheKey(req) {
-  if (req.remoteUser) {
+  // If it's not a GraphQL query with such properties, no cache
+  if (!req.body || !req.body.operationName || !req.body.query || !req.body.variables) {
     return;
   }
-  if (!req.body || !req.body.operationName || !req.body.query) {
+
+  // We only want to cache queries with an Account slug
+  // TODO: check that slug never express something else like a Tier
+  const slug = req.body.variables.slug || req.body.variables.collectiveSlug;
+  if (!slug) {
```

Comment:
Out of safety, we should ignore queries that contain variables that give you special permissions, such as `draftKey` for expenses (though I acknowledge it has no chance to happen with the code as it is today since the `ExpenseQuery` doesn't take a slug as argument).

I don't have any other cases in mind right now, but we can also be defensive and protect against anything that sounds like a private variable (e.g. `token`, `password`, etc). Maybe something to synchronize with https://github.com/opencollective/opencollective-api/pull/8452?